firebase-sessions/src/main/kotlin/com/google/firebase/sessions/WallClock.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.sessions
+
+import android.os.Build
+import android.os.SystemClock
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.nanoseconds
+
+/** Util object for "wall clock" time functions. */
+internal object WallClock {
+  /** Gets the [Duration] elapsed in "wall clock" time since device boot. */
+  fun elapsedRealtime(): Duration =
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+      SystemClock.elapsedRealtimeNanos().nanoseconds
```

Comment:
For easier understanding, can we convert the nanoSeconds here to milliseconds? That way, it becomes clear in the code that both are represented as duration of time with reference of milliseconds level precision captured from the SystemClock?

Reply:
Done. And since we will always be millis precision, we can just use the older api for all apl levels.